src/shogun/lib/SGVector.cpp

```diff
@@ -371,7 +370,7 @@ template <>
 void SGVector<bool>::display_vector(const bool* vector, int32_t n, const char* name,
 		const char* prefix)
 {
-	ASSERT(n>=0)
+	REQUIRE(n>=0, "Vector size can not be negative.\n");
```

Comment:
Nah, never mind about that one. It should be understandable enough to say that a negative value was passed.